tests/integration/test_charm.py

```diff
@@ -15,28 +17,115 @@
 METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
 APP_NAME = METADATA["name"]
 
+UNIT_IDS = [0, 1, 2]
 
+
+@pytest.mark.skip_if_deployed
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test: OpsTest):
-    """Build the charm-under-test and deploy it together with related charms.
+    """Build the charm-under-test and deploy it.
 
     Assert on the unit status before any relations/configurations take place.
     """
     # build and deploy charm from local source folder
     charm = await ops_test.build_charm(".")
     resources = {"mysql-image": METADATA["resources"]["mysql-image"]["upstream-source"]}
-    await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME)
+    await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME, num_units=1)
 
     # issuing dummy update_status just to trigger an event
-    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+    await ops_test.model.set_config({"update-status-hook-interval": "60s"})
 
     await ops_test.model.wait_for_idle(
         apps=[APP_NAME],
         status="active",
         raise_on_blocked=True,
         timeout=1000,
     )
+    assert len(ops_test.model.applications[APP_NAME].units) == 1
     assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
 
-    # effectively disable the update status from firing
-    await ops_test.model.set_config({"update-status-hook-interval": "60m"})
+
+@pytest.mark.skip_if_deployed
+@pytest.mark.abort_on_fail
+async def test_scale_to_ha(ops_test: OpsTest):
+    """Scale the charm to HA."""
+    await ops_test.model.applications[APP_NAME].scale(3)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
+    assert len(ops_test.model.applications[APP_NAME].units) == 3
```

Comment:
Maybe it might be worth checking if they are part of the same cluster or something else to ensure they scale and provide HA in this test. I know the following test guarantees this, but I'm wondering if it's not better to test something else here to justify the test's name.

Reply:
You're right. Will check that.